daemon/daemon.go

```diff
@@ -525,21 +525,34 @@ func NewDaemon(c *Config) (*Daemon, error) {
 
 	var kvClient kvstore.KVClient
 
-	// FIXME: This should really be a single configuration flag
-	if c.ConsulConfig != nil {
-		c, err := kvstore.NewConsulClient(c.ConsulConfig)
-		if err != nil {
-			return nil, err
+	// Create new key-value store client structures based on provided kvstore type.
+	switch c.KVStore {
+	case kvstore.Consul:
+		if c.ConsulConfig != nil {
+			log.Infof("Using consul as key-value store")
+			c, err := kvstore.NewConsulClient(c.ConsulConfig)
+			if err != nil {
+				return nil, err
+			}
+			kvClient = c
+		} else {
+			log.Warningf("Invalid configuration for consul provided; using local storage.")
```

Comment:
I think we should error out in this case with a message such as "Please specify the address to a consul instance with the --consul option" as the user explicitly asked for a consul kvstore.